src/Microsoft.AspNet.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -36,8 +36,7 @@ public abstract class RemoteAuthenticationHandler<TOptions> : AuthenticationHand
                     return false;
                 }
 
-                Context.Response.StatusCode = 500;
-                return true;
+                throw errorContext.Error;
```

Comment:
Not sure I follow, the errors in context today aren't actually usually caught, we are constructing the exceptions mostly as opposed to catching and stuffing them in here...